lib/proxy/signal_handler.rb

```diff
@@ -14,4 +23,22 @@ def self.install_ttin_trap
       end
     end
   end
+
+  def install_int_trap
+    if Rack.release < '1.6.4'
+      # Rack installs its own trap; Sleeping for 5 secs insures we overwrite it with our own
+      Thread.new do
+        sleep 5
+        begin
+          trap(:INT) { exit(0) }
+        rescue Exception => e
+          logger.warn "Unable to overwrite interrupt trap: #{e}"
+        end
+      end
+    end
+  end
+
+  def install_term_trap
+    trap(:TERM) { exit(0) }
+  end
```

Comment:
Shell we rather signal shutdown to our worker threads rather than hard exiting this? I was browsing through the Rack code and I don't know how to tell the stack to cleanly exit. Maybe it's not possible and we just need to do this, but it is worth some effort. Broken orchestrations are always problematic.